thorlcr/activities/indexwrite/thindexwrite.cpp

```diff
@@ -279,6 +279,22 @@ class IndexWriteActivityMaster : public CMasterActivity
                 VStringBuffer pval("%f", info->getBloomProbability());
                 bloom->setProp("@bloomProbability", pval.str());
             }
+            // Update numDiskWrites
+            StatsScopeId rootScope(SSTfile, 0U);
+            Owned<IStatisticGatherer> fileStats;
+            rootScope.setFileId(fileName);
+            fileStats.setown(createStatisticsGatherer(SCTall, "", rootScope));
```

Comment:
A couple of issues.  You would normally want to use a CRuntimeStatisticCollection to serialize information from a slave to the master.  the createStatisticsGatherer() is generally used once the stats are going to be published.
Also SCTall is not correct - in this case it would probably be SCTthor.
I also agree with Jake's comment below.

Reply:
@jakesmith I'm slightly surprised the information isn't already serialized in the activity stats.